plugins/jobs/src/js/pages/JobRunHistoryTable.js

```diff
@@ -365,65 +346,78 @@ class JobRunHistoryTable extends React.Component {
     return <div>{runTimeFormat}</div>;
   }
 
+  renderActionsColumn(prop, row) {
+    if (!["ACTIVE", "INITIAL", "STARTING"].includes(row.status)) {
+      return;
+    }
+
+    const actions = [
+      {
+        className: "hidden",
+        id: "default",
+        html: "",
+        selectedHtml: (
+          <Icon shape={SystemIcons.EllipsisVertical} size={iconSizeXs} />
+        )
+      },
+      {
+        id: STOP,
+        html: <Trans render="span" className="text-danger" id={STOP} />
+      }
+    ];
+
+    return (
+      <Dropdown
+        anchorRight={true}
+        buttonClassName="button button-mini button-link"
+        dropdownMenuClassName="dropdown-menu"
+        dropdownMenuListClassName="dropdown-menu-list"
+        dropdownMenuListItemClassName="clickable"
+        wrapperClassName="dropdown flush-bottom table-cell-icon actions-dropdown"
+        items={actions}
+        persistentID={"default"}
+        onItemSelection={this.handleItemSelect.bind(this, row.id)}
+        scrollContainer=".gm-scroll-view"
+        scrollContainerParentSelector=".gm-prevented"
+        transition={true}
+        transitionName="dropdown-menu"
+        disabled={false}
+      />
+    );
+  }
+
   renderTimeColumn(prop, row) {
     // L10NTODO: Relative time
     return <TimeAgo time={row[prop]} autoUpdate={false} />;
   }
 
   render() {
     const { job, i18n } = this.props;
-    let { checkedItems } = this.state;
-    const disabledItems = this.getDisabledItemsMap(job);
+    const { selectedID } = this.state;
     const totalRunCount = job.jobRuns.nodes.length;
-    let rightAlignLastNChildren = 0;
-    let hasCheckedTasks = false;
-
-    // Remove all disabled items from the checkedItems.
-    checkedItems = Object.keys(checkedItems).reduce(function(
-      filteredItems,
-      key
-    ) {
-      if (!disabledItems[key]) {
-        filteredItems[key] = checkedItems[key];
-        hasCheckedTasks = true;
-      }
-
-      return filteredItems;
-    },
-    {});
-
-    if (hasCheckedTasks) {
-      rightAlignLastNChildren = 1;
+    const disabledIDs = this.getDisabledItemsMap(job);
+    if (disabledIDs[selectedID]) {
+      this.handleItemSelect(null);
```

Comment:
what does this do? (i don't mean codewise - i mean "what user story is covered by this"?)

Reply:
The same we did before with https://github.com/dcos/dcos-ui/pull/4069/files/99262bc4d88b14e945ae5149c516d17bfc532604#diff-40f057127c4e7bc0ed23cfdca70e59b3L381 , but for a single ID - we remove it from selectedItem.

Reply:
why would that be necessary now that we only have a single id? is `render` the right place to do this?

Reply:
It's necessary, since the function that should do this https://github.com/dcos/dcos-ui/pull/4069/files#diff-816c1e88efffe7eed722c8faa498cd04R49 doesn't work. It used to be in render and it is still in render. It need a refactoring, but not in this PR.